### PR TITLE
Add configurable docker credentials

### PR DESCRIPTION
**Pull request by manuel-sanchez**
Add the ability to specify a docker conf secret to pull a custom image of external secrets

### CODE REVIEW

File: charts/kubernetes-external-secrets/templates/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: {{ template "kubernetes-external-secrets.serviceAccountName" . }}
+      {{- if .Values.image.imagePullSecrets }}
+      imagePullSecrets:
+        - name: {{ .Values.image.imagePullSecrets }}
```
Comment (Flydiverny) [inline charts/kubernetes-external-secrets/templates/deployment.yaml]:
```suggestion
        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
```

not sure if the 8 is correct, please validate 😅

File: charts/kubernetes-external-secrets/values.yaml [hunk 2]
```diff
@@ -44,6 +44,7 @@ image:
   repository: godaddy/kubernetes-external-secrets
   tag: 2.2.1
   pullPolicy: IfNotPresent
+  imagePullSecrets: ""
```
Comment (Flydiverny) [inline charts/kubernetes-external-secrets/values.yaml]:
```suggestion

imagePullSecrets: []
```

moving it toplevel

File: charts/kubernetes-external-secrets/templates/deployment.yaml [hunk 3]
```diff
@@ -27,6 +27,10 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: {{ template "kubernetes-external-secrets.serviceAccountName" . }}
+      {{- if .Values.image.imagePullSecrets }}
```
Comment (Flydiverny) [inline charts/kubernetes-external-secrets/templates/deployment.yaml]:
```suggestion
      {{- if .Values.imagePullSecrets }}
```

**Comment by manuel-sanchez** [comment]
Changes and README updated!

**Pull request merged**